internal/featureflag/cache.go

```diff
@@ -0,0 +1,73 @@
+package featureflag
+
+import (
+	"errors"
+	"fmt"
+
+	"github.com/gomodule/redigo/redis"
+
+	"github.com/sourcegraph/sourcegraph/internal/actor"
+	"github.com/sourcegraph/sourcegraph/internal/redispool"
+)
+
+var (
+	pool = redispool.Store
+)
+
+func getEvaluatedFlagSetFromCache(flagsSet *FlagSet) EvaluatedFlagSet {
+	evaluatedFlagSet := EvaluatedFlagSet{}
+
+	c := pool.Get()
+	defer c.Close()
+
+	visitorID, err := getVisitorIDForActor(flagsSet.actor)
+
+	if err != nil {
+		return evaluatedFlagSet
+	}
+
+	for k := range flagsSet.flags {
+		if value, err := redis.Bool(c.Do("HGET", getFlagCacheKey(k), visitorID)); err == nil {
+			evaluatedFlagSet[k] = value
+		}
+	}
+
+	return evaluatedFlagSet
+}
+
+func setEvaluatedFlagToCache(a *actor.Actor, flagName string, value bool) {
+	c := pool.Get()
+	defer c.Close()
+
+	var visitorID string
+
+	visitorID, err := getVisitorIDForActor(a)
+
+	if err != nil {
+		return
+	}
+
+	c.Do("HSET", getFlagCacheKey(flagName), visitorID, fmt.Sprintf("%v", value))
```

Comment:
We can be a bit more precise about the fact that we require a bool here:
```suggestion
	c.Do("HSET", getFlagCacheKey(flagName), visitorID, strconv.FormatBool(value))
```